eras/dijkstra/impl/testlib/Test/Cardano/Ledger/Dijkstra/Binary/Golden.hs

```diff
@@ -1,27 +1,83 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 
 module Test.Cardano.Ledger.Dijkstra.Binary.Golden (
-  goldenListRedeemersDisallowed,
+  spec,
 ) where
 
 import Cardano.Ledger.Alonzo.TxWits (Redeemers)
-import Cardano.Ledger.Binary (DecoderError (..), DeserialiseFailure (..))
-import Cardano.Ledger.Dijkstra.Core (eraProtVerLow)
-import Test.Cardano.Ledger.Common (Spec, it)
-import Test.Cardano.Ledger.Conway.Binary.Golden (expectDecoderFailure, listRedeemersEnc)
+import Cardano.Ledger.BaseTypes (Version)
+import Cardano.Ledger.Binary (DecoderError (..), DeserialiseFailure (..), Tokens (..))
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Conway.TxCert (Delegatee (..))
+import Cardano.Ledger.Credential (Credential (..))
+import Cardano.Ledger.Dijkstra.Core (
+  EraTxBody (..),
+  EraTxOut (..),
+  TxLevel (..),
+  eraProtVerLow,
+  pattern DelegTxCert,
+ )
+import Cardano.Ledger.TxIn (TxIn (..))
+import qualified Data.Set as Set
+import Test.Cardano.Ledger.Binary.Plain.Golden (Enc (..))
+import Test.Cardano.Ledger.Common (Spec, describe, it)
+import Test.Cardano.Ledger.Conway.Binary.Golden (expectDecoderFailureAnn, listRedeemersEnc)
+import Test.Cardano.Ledger.Core.KeyPair (mkKeyHash)
 import Test.Cardano.Ledger.Dijkstra.Era (DijkstraEraTest)
 
+spec :: forall era. DijkstraEraTest era => Spec
+spec = describe "Golden" $ do
+  goldenListRedeemersDisallowed @era
+  goldenDuplicateCertsDisallowed @era
+
 goldenListRedeemersDisallowed :: forall era. DijkstraEraTest era => Spec
 goldenListRedeemersDisallowed =
   it "Decoding Redeemers encoded as a list fails" $
-    expectDecoderFailure @(Redeemers era)
+    expectDecoderFailureAnn @(Redeemers era)
       (eraProtVerLow @era)
       listRedeemersEnc
       ( DecoderErrorDeserialiseFailure
           "Annotator (MemoBytes (RedeemersRaw DijkstraEra))"
           (DeserialiseFailure 0 "List encoding of redeemers not supported starting with PV 12")
       )
+
+duplicateCertsTx :: forall era. DijkstraEraTest era => Version -> Enc
```

Comment:
I think it would be an awesome addition if we moved this example to shelley era and added it as a positive test, where:
* In Shelley till Babbage we successfully decode such transaction, but only one certificate is preserved
* In Conway only one certificate is preserved